README.md
# React Native Expo Starter UI Kit
![Adsız (950 × 600 piksel) (900 × 600 piksel) (2)](https://user-images.githubusercontent.com/38071168/166121208-65f95ee4-aed3-401e-b112-6028b5cd6d43.png)


### Project Installation Instructions
The project was developed via Expo CLI.

```
  $ npm install -g expo-cli

  $ git clone https://github.com/AyberkCakar/react-native-expo-starter-kit.git
  
  # package install
  $ npm install (or) yarn install (or) expo install

  # project start
  $ expo start
  
```
### Project Description
The project was developed via Expo CLI. The ready-made mobile UI design was based and improvements were made on it. In practice, the firebase (auth, firestore, storage) infrastructure was used. In the application, there are UI components, map, push notification and database infrastructure. The application will continue to be improved. The application will continue to be improved. You can view changes made to the application in the ChangeLog or under projects.

### Pages Within the Project

```
  - Sign In
  - Sign Up
  - Home
  - Profile
  - Edit User
  - Users
  - Component Examples
  - Notification
  - Notifications
  - Map
```


### Using Firebase
```
  Auth
    - Sign In
    - Sign Up
  
  Firestore
    - User Information
    - Notifications
    
  Storage 
    - User Image
```

### Firebase Schemas

##### User Schema
```
  uid?: string;
  name: string;
  email: string;
  twitter?: string;
  instagram?: string;
  facebook?: string;
  github?: string;
  title?: string;
  aboutMe?: string;
  image?: string;
  repos?: number;
  followers?: number;
  following?: number;
  company?: string;
  ```
  
  ##### Notification Schema
```   
  uid: string;
  is_read?: boolean;
  is_image?: boolean;
  description?: string;
  title?: string;
  image?: string;
  user_uid?: string;
  detail?: string;
  detail_image?: string;
  created_at?: Date;
  ```

### Related Articles
* [React Native - Expo Neden Kullanmalıyım?](https://ayberkcakar.medium.com/react-native-expo-neden-kullanmal%C4%B1y%C4%B1m-394235e8c9d6)
* [React Native - Expo Nasıl Kurabilirim?](https://ayberkcakar.medium.com/react-native-expo-nas%C4%B1l-kurabilirim-3fe178a992f6)


## Useful resources
* [Changelog](CHANGELOG.md)


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

## [1.9.0](https://github.com/AyberkCakar/react-native-expo-starter-kit/compare/v1.8.0...v1.9.0) (2022-04-26)


### Features

* added a firebase connection to the notifications page ([ac1686d](https://github.com/AyberkCakar/react-native-expo-starter-kit/commit/ac1686defd84b0462aafb468587b85d65b393e56))
* added delete all notifications ([b0a7fb8](https://github.com/AyberkCakar/react-native-expo-starter-kit/commit/b0a7fb8ab837b46d62fdb7bda40779c3ccf619e1))
* added firebase storage ([d91e378](https://github.com/AyberkCakar/react-native-expo-starter-kit/commit/d91e378eb1b2250d6111427dea8ff4f57e1e03e2))
* added mark all read ([e84c230](https://github.com/AyberkCakar/react-native-expo-starter-kit/commit/e84c230a5e2df458d11365d0158ddf639da718f3))
* added notification date show difference ([50923fb](https://github.com/AyberkCakar/react-native-expo-starter-kit/commit/50923fb7600e0338cad7ac29e8215d78bbf802cc))
* added notification detail data connection ([d49f7b5](https://github.com/AyberkCakar/react-native-expo-starter-kit/commit/d49f7b56c94ccd56cd55e4f3e4b63f8a99e2a9f2))
* added notification detail page ([d8499ee](https://github.com/AyberkCakar/react-native-expo-starter-kit/commit/d8499ee64816730fc56f0932052436fbd76fe7f9))
* added notification interface ([1ec0806](https://github.com/AyberkCakar/react-native-expo-starter-kit/commit/1ec0806806948bf7631df44414983bab5abd36d1))
* added notification read (is_read update) ([5e56b6a](https://github.com/AyberkCakar/react-native-expo-starter-kit/commit/5e56b6aac8163c1cf1c3aeab2e01a8ba88e9510c))
* added notifications page ([f73202e](https://github.com/AyberkCakar/react-native-expo-starter-kit/commit/f73202ed3787bfb7931cab9dcb964896f65fef5c))
* added your profile has been viewed notification has been. ([addb20b](https://github.com/AyberkCakar/react-native-expo-starter-kit/commit/addb20be880c7af54052882df43694432e4f00ec))


### Bug Fixes

* fixed not send github request ([1566037](https://github.com/AyberkCakar/react-native-expo-starter-kit/commit/15660370054c02c2873c0c092ab1e7718e4df41e))
* fixed notification card design ([07626e1](https://github.com/AyberkCakar/react-native-expo-starter-kit/commit/07626e1d314a1cf72329751716990fe24d4eee20))
* fixed short date name ([e33af52](https://github.com/AyberkCakar/react-native-expo-starter-kit/commit/e33af52d68bf21cfac83843937257c502aa72cc9))
* fixed user information texts color ([4e03c9c](https://github.com/AyberkCakar/react-native-expo-starter-kit/commit/4e03c9c27845bbbf1237484148b962113a0e748b))

## [1.8.0](https://github.com/AyberkCakar/react-native-expo-starter-kit/compare/v1.7.0...v1.8.0) (2022-04-24)


### Features

* added a firebase connection to the users page ([d993d1d](https://github.com/AyberkCakar/react-native-expo-starter-kit/commit/d993d1d51faf66e6ca126aab828e061f5dcc8d23))
* added edit user firebase connection ([2b5b0aa](https://github.com/AyberkCakar/react-native-expo-starter-kit/commit/2b5b0aa781fcd2ca195ff0429e33e999b0bb3c94))
* added edit user page ([09ef612](https://github.com/AyberkCakar/react-native-expo-starter-kit/commit/09ef6128db1743616deb683ec01ee80d9928f854))
* added github api in profile page ([5237dda](https://github.com/AyberkCakar/react-native-expo-starter-kit/commit/5237ddac0b3412d0e63959445ecb9697e3940716))
* added image upload component and firebase connection ([68e24c5](https://github.com/AyberkCakar/react-native-expo-starter-kit/commit/68e24c5a290f17b9dc7db01819d0b0efe8a7a485))
* added notification service ([5ab718d](https://github.com/AyberkCakar/react-native-expo-starter-kit/commit/5ab718dbe90a9922c5022fdadde3e9ccf5d1625b))
* added notification settings in app.json ([08bf4e8](https://github.com/AyberkCakar/react-native-expo-starter-kit/commit/08bf4e896976f962c28081a2a705c701b6d805cb))
* added profile page ([c96fcf9](https://github.com/AyberkCakar/react-native-expo-starter-kit/commit/c96fcf9b0a0881d4659c956174a58fedc3b72af6))
* added profile page connection with user store ([1ab65e8](https://github.com/AyberkCakar/react-native-expo-starter-kit/commit/1ab65e8bb34f5675f03eb9b2cfc4826adf552e20))
* added push notification for post registration ([38144b0](https://github.com/AyberkCakar/react-native-expo-starter-kit/commit/38144b00ffcacc512352c94bfd9d453c20ddf0fe))
* added Storage service ([54338ad](https://github.com/AyberkCakar/react-native-expo-starter-kit/commit/54338ad478be7d8a675ca9718649b1ace2cda256))
* added toast message in edit user ([fa85260](https://github.com/AyberkCakar/react-native-expo-starter-kit/commit/fa85260628f677fbfe693d4141943a0d993a7280))
* added toast package ([8600cb4](https://github.com/AyberkCakar/react-native-expo-starter-kit/commit/8600cb45c8768d4900d68b6bd957e06ea3558be1))
* added user search ([675ff73](https://github.com/AyberkCakar/react-native-expo-starter-kit/commit/675ff73c39aa52ddc1487cd6436095dd6fa2637c))
* added users page ([e485c90](https://github.com/AyberkCakar/react-native-expo-starter-kit/commit/e485c90667044646e122df41d57089d681133cca))
* added users page and profile page navigation link ([d9f1d2a](https://github.com/AyberkCakar/react-native-expo-starter-kit/commit/d9f1d2a9433210caf7e97773136ad3b5de58a38d))


### Bug Fixes

* fixed company field check ([a337759](https://github.com/AyberkCakar/react-native-expo-starter-kit/commit/a337759415e9c5e95e6e9599ff2afdafde42e2f3))
* fixed not refreshing when coming back to profile page ([c3ba915](https://github.com/AyberkCakar/react-native-expo-starter-kit/commit/c3ba915b042f1758e721fd816e379acf834bf53e))
* fixed search not clearing on page transitions ([0419a60](https://github.com/AyberkCakar/react-native-expo-starter-kit/commit/0419a60c237a62183ef93d3951bbdcf34dfb12af))

src/constants/types/theme.ts
import * as React from 'react';
import {
  ColorValue,
  FlexStyle,
  ImageSourcePropType,
  ScaledSize,
  TextStyle,
} from 'react-native';

// Spacing types
export interface ISpacing
  extends Pick<
    FlexStyle,
    | 'margin'
    | 'marginVertical'
    | 'marginHorizontal'
    | 'marginLeft'
    | 'marginRight'
    | 'marginTop'
    | 'marginBottom'
    | 'padding'
    | 'paddingVertical'
    | 'paddingHorizontal'
    | 'paddingLeft'
    | 'paddingRight'
    | 'paddingTop'
    | 'paddingBottom'
  > {}

export type TWeight =
  /** fontWeight: 400 */
  | 'normal'
  /** fontWeight: 100 */
  | 'thin'
  /** fontWeight: 200 */
  | 'extralight'
  /** fontWeight: 300 */
  | 'light'
  /** fontWeight: 500 */
  | 'medium'
  /** fontWeight: 600 */
  | 'semibold'
  /** fontWeight: 700 */
  | 'bold'
  /** fontWeight: 800 */
  | 'extrabold'
  /** fontWeight: 900 */
  | 'black';

export interface ITheme {
  colors: ThemeColors;
  gradients: ThemeGradients;
  sizes: ThemeSizes & ThemeSpacing & ICommonTheme['sizes'];
  assets: ThemeAssets & ThemeIcons;
  icons: ThemeIcons;
  fonts: ThemeFonts;
  weights: ThemeWeights;
  lines: ThemeLineHeights;
}
export interface ICommonTheme {
  assets: ThemeAssets & ThemeIcons;
  icons: ThemeIcons;
  fonts: ThemeFonts;
  weights: ThemeWeights;
  lines: ThemeLineHeights;
  sizes: {
    width: ScaledSize['width'];
    height: ScaledSize['height'];
  };
}

export interface IThemeProvider {
  children?: React.ReactNode;
  theme?: ITheme;
  setTheme?: (theme?: ITheme) => void;
}

export interface ThemeColors {
  text: ColorValue;
  primary: ColorValue;
  secondary: ColorValue;
  tertiary: ColorValue;
  black: ColorValue;
  white: ColorValue;
  light: ColorValue;
  dark: ColorValue;
  gray: ColorValue;
  danger: ColorValue;
  warning: ColorValue;
  success: ColorValue;
  info: ColorValue;
  card: ColorValue;
  background: ColorValue;
  shadow: ColorValue;
  overlay: ColorValue;
  focus: ColorValue;
  input: ColorValue;
  switchOn: ColorValue;
  switchOff: ColorValue;
  checkbox: string[];
  checkboxIcon: ColorValue;
  facebook: ColorValue;
  twitter: ColorValue;
  dribbble: ColorValue;
  github: ColorValue;
  instagram: ColorValue;
  icon: ColorValue;
  blurTint: 'light' | 'dark' | 'default';
  link: ColorValue;
}

export interface ThemeGradients {
  primary?: string[];
  secondary?: string[];
  tertiary?: string[];
  black?: string[];
  white?: string[];
  light?: string[];
  dark?: string[];
  gray?: string[];
  danger?: string[];
  warning?: string[];
  success?: string[];
  info?: string[];
  divider?: string[];
  menu?: string[];
}

export interface ThemeSizes {
  base: number;
  text: number;
  radius: number;
  padding: number;

  h1: number;
  h2: number;
  h3: number;
  h4: number;
  h5: number;
  p: number;

  buttonBorder: number;
  buttonRadius: number;
  socialSize: number;
  socialRadius: number;
  socialIconSize: number;

  inputHeight: number;
  inputBorder: number;
  inputRadius: number;
  inputPadding: number;

  shadowOffsetWidth: number;
  shadowOffsetHeight: number;
  shadowOpacity: number;
  shadowRadius: number;
  elevation: number;

  cardRadius: number;
  cardPadding: number;

  imageRadius: number;
  avatarSize: number;
  avatarRadius: number;

  switchWidth: number;
  switchHeight: number;
  switchThumb: number;

  checkboxWidth: number;
  checkboxHeight: number;
  checkboxRadius: number;
  checkboxIconWidth: number;
  checkboxIconHeight: number;

  linkSize: number;

  multiplier: number;
}

export interface ThemeSpacing {
  xs: number;
  s: number;
  sm: number;
  m: number;
  md: number;
  l: number;
  xl: number;
  xxl: number;
}

export interface ThemeWeights {
  text: TextStyle['fontWeight'];
  h1?: TextStyle['fontWeight'];
  h2?: TextStyle['fontWeight'];
  h3?: TextStyle['fontWeight'];
  h4?: TextStyle['fontWeight'];
  h5?: TextStyle['fontWeight'];
  p?: TextStyle['fontWeight'];

  thin: TextStyle['fontWeight'];
  extralight: TextStyle['fontWeight'];
  light: TextStyle['fontWeight'];
  normal: TextStyle['fontWeight'];
  medium: TextStyle['fontWeight'];
  semibold?: TextStyle['fontWeight'];
  bold?: TextStyle['fontWeight'];
  extrabold?: TextStyle['fontWeight'];
  black?: TextStyle['fontWeight'];
}
export interface ThemeIcons {
  apple: ImageSourcePropType;
  google: ImageSourcePropType;
  facebook: ImageSourcePropType;
  arrow: ImageSourcePropType;
  articles: ImageSourcePropType;
  basket: ImageSourcePropType;
  bell: ImageSourcePropType;
  calendar: ImageSourcePropType;
  chat: ImageSourcePropType;
  check: ImageSourcePropType;
  clock: ImageSourcePropType;
  close: ImageSourcePropType;
  components: ImageSourcePropType;
  document: ImageSourcePropType;
  documentation: ImageSourcePropType;
  extras: ImageSourcePropType;
  flight: ImageSourcePropType;
  home: ImageSourcePropType;
  hotel: ImageSourcePropType;
  image: ImageSourcePropType;
  location: ImageSourcePropType;
  menu: ImageSourcePropType;
  more: ImageSourcePropType;
  notification: ImageSourcePropType;
  office: ImageSourcePropType;
  payment: ImageSourcePropType;
  profile: ImageSourcePropType;
  register: ImageSourcePropType;
  rental: ImageSourcePropType;
  search: ImageSourcePropType;
  settings: ImageSourcePropType;
  star: ImageSourcePropType;
  train: ImageSourcePropType;
  users: ImageSourcePropType;
  warning: ImageSourcePropType;
}

export interface ThemeAssets {
  OpenSansLight?: any;
  OpenSansRegular?: any;
  OpenSansSemiBold?: any;
  OpenSansExtraBold?: any;
  OpenSansBold?: any;

  logo: ImageSourcePropType;
  header: ImageSourcePropType;
  background: ImageSourcePropType;

  card1: ImageSourcePropType;
  card2: ImageSourcePropType;
  card3: ImageSourcePropType;
  card4: ImageSourcePropType;
  card5: ImageSourcePropType;

  photo1: ImageSourcePropType;
  photo2: ImageSourcePropType;
  photo3: ImageSourcePropType;
  photo4: ImageSourcePropType;
  photo5: ImageSourcePropType;
  photo6: ImageSourcePropType;
  carousel1: ImageSourcePropType;

  avatar1: ImageSourcePropType;
  avatar2: ImageSourcePropType;
  anonymous: ImageSourcePropType;

  x5: ImageSourcePropType;
  gle: ImageSourcePropType;
  tesla: ImageSourcePropType;

  ios: ImageSourcePropType;
  android: ImageSourcePropType;

  tr: ImageSourcePropType;
  en: ImageSourcePropType;
}

export interface ThemeFonts {
  text: string;
  h1: string;
  h2: string;
  h3: string;
  h4: string;
  h5: string;
  p: string;
  thin: string;
  extralight: string;
  light: string;
  normal: string;
  medium: string;
  bold: string;
  semibold: string;
  extrabold: string;
  black: string;
}

export interface ThemeLineHeights {
  text: number;
  h1: number;
  h2: number;
  h3: number;
  h4: number;
  h5: number;
  p: number;
}
